widget/entry.go

```diff
@@ -95,17 +93,20 @@ func (e *Entry) CreateRenderer() fyne.WidgetRenderer {
 	cursor := canvas.NewRectangle(theme.FocusColor())
 	cursor.Hide()
 
-	objects := []fyne.CanvasObject{line, e.placeholderProvider(), e.textProvider(), cursor}
+	var objects []fyne.CanvasObject
+	e.SetFields(func() {
```

Comment:
Why do you need the Write lock to read line, e.placeholderProvider, e.textProvider, and cursor?

Reply:
You don't this was just a shorthand to encapsulate a read, then possible write, then more reads. I can break it out into separate areas but the code will get rather ugly.